providers/implementations/keymgmt/dsa_kmgmt.c

```diff
@@ -109,7 +109,7 @@ static int params_to_key(DSA *dsa, const OSSL_PARAM params[])
     return 1;
 
  err:
-    BN_free(priv_key);
+    BN_clear_free(priv_key);
```

Comment:
This is technically unrelated to the rest of the PR, should the commit with this change for dh and dsa go in a separate PR?